### PR TITLE
ENH: stats: Implement _logpdf, _sf and _isf for nakagami.

### PR DESCRIPTION
**Pull request by WarrenWeckesser**

### CODE REVIEW

**Review by ev-br — COMMENTED** [review]
LGTM modulo a small nit

**Comment by WarrenWeckesser** [comment]
Thanks, @ev-br; what is the small nit?   I don't see any other comments.

File: scipy/stats/_continuous_distns.py [hunk 1]
```diff
@@ -5550,16 +5550,26 @@ class nakagami_gen(rv_continuous):
...
     """
     def _pdf(self, x, nu):
+        return np.exp(self._logpdf(x, nu))
+
+    def _logpdf(self, x, nu):
         # nakagami.pdf(x, nu) = 2 * nu**nu / gamma(nu) *
         #                       x**(2*nu-1) * exp(-nu*x**2)
-        return 2*nu**nu/sc.gamma(nu)*(x**(2*nu-1.0))*np.exp(-nu*x*x)
+        return (np.log(2) + nu*np.log(nu) - sc.gammaln(nu) +
+                (2*nu - 1)*np.log(x) - nu*x**2)
```
Comment (ev-br) [inline scipy/stats/_continuous_distns.py:5559]: Nit, optional: `xlogy(2*nu -1, x)`

**Comment by ev-br** [comment]
Hm, I seem to be unable to use github today. Here's the small optional nit: 
https://github.com/scipy/scipy/pull/12958#discussion_r509465051 
In case I cannot even copy-paste links: it's about using `xlogy(2*nu - 1, x)` in place of `(2*nu -1) * log(x)`

File: scipy/stats/_continuous_distns.py (hunk 1 above)
Comment (WarrenWeckesser) [reply scipy/stats/_continuous_distns.py]: OK, I've changed that, and I've changed `nu*np.log(nu)` to `xlogy(nu, nu)`.

**Comment by ev-br** [comment]
Merged, thanks Warren.

**Pull request merged**